src/app/static/src/tests/components/projects/shareProject.test.ts

```diff
@@ -628,4 +628,91 @@ describe("ShareProject", () => {
         expect(mockTooltip.mock.calls[0][1].value).toBe("Partager");
     });
 
+    it("if email entered is valid, cycles to empty input then ok button on enter presses", async (done) => {
+        const cloneProject = jest.fn();
+        const wrapper = mount(ShareProject, {
+            attachToDocument: true,
+            propsData: {
+                project: {id: 1, name: "p1"}
+            },
+            store: createStore(jest.fn().mockResolvedValue(true), cloneProject),
+        });
+
+        const link = wrapper.find("button");
+        link.trigger("click");
+        const input = wrapper.find(Modal).find("input");
+        input.trigger("focus")
+        input.setValue("goodemail");
+        input.trigger("keyup.enter");
+
+        setTimeout(() => {
+            const modal = wrapper.find(Modal);
+            expect(modal.find("input").classes()).not.toContain("is-invalid");
+            expect(modal.find(".text-danger").exists()).toBe(false);
+            expect(wrapper.find(Modal).findAll("input").length).toBe(2);
+
+            const input2 = wrapper.find(Modal).findAll("input").at(1);
+            expect(input2.element).toBe(document.activeElement);
+            input2.trigger("keyup.enter");
+
+            setTimeout(() => {
+                const okBtn = modal.find("button")
+                expect(okBtn.element).toBe(document.activeElement);
+                expect(okBtn.attributes("disabled")).toBeUndefined();
+                expect(modal.find(".help-text").isVisible()).toBe(false);
+                // okBtn.trigger("keyup.enter");
+                okBtn.trigger("click");
```

Comment:
Ideally we'd trigger an enter press here but this doesn't do anything in vue tests; however, in the browser, pressing enter on an active element seems to produce the same behaviour as clicking it

Reply:
Out of curiosity, is this issue specific to buttons, rather than text inputs?

Reply:
Yes, this is a button specific issue. Vue tests recognise keyup triggers on inputs.

Reply:
Ok, thanks. Best to add a comment in the code explaining this, as it probably won't be obvious to the whoever might be editing these tests in the future.